ethportal-api/src/types/portal_wire.rs

```diff
@@ -157,117 +157,68 @@ pub enum DiscoveryRequestError {
     InvalidMessage,
 }
 
-#[derive(Error, Debug)]
-pub enum ProtocolIdError {
-    #[error("Unable to decode protocol id to bytes")]
-    Decode(ByteUtilsError),
-
-    #[error("invalid protocol id")]
-    Invalid,
-}
-
-/// Protocol identifiers
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
-pub enum ProtocolId {
-    State,
-    VerkleState,
-    History,
-    TransactionGossip,
-    CanonicalIndices,
-    Beacon,
-    Utp,
-}
-
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct NetworkSpec {
     network: Network,
-    portal_networks: BiHashMap<ProtocolId, String>,
+    // mapping of subnetworks to protocol hex id strings
+    portal_subnetworks: BiHashMap<Subnetwork, String>,
 }
 
 impl NetworkSpec {
     pub fn network(&self) -> Network {
         self.network
     }
 
-    pub fn get_protocol_id_from_hex(&self, hex: &str) -> Result<ProtocolId, ProtocolIdError> {
-        self.portal_networks
+    pub fn get_subnetwork_from_hex(&self, hex: &str) -> Result<Subnetwork, SubnetworkError> {
+        self.portal_subnetworks
             .get_by_right(hex)
             .copied()
-            .ok_or(ProtocolIdError::Invalid)
+            .ok_or(SubnetworkError::Invalid)
     }
 
-    pub fn get_protocol_hex_from_id(
+    pub fn get_hex_from_subnetwork(
```

Comment:
```suggestion
    pub fn get_protocol_identifier_from_subnetwork(
```

Reply:
I am just confused what the hex is, which is why I mention it. I think by saying `protocol_indentifier` instead of hex we more clearly link this to what it is in the spec

Reply:
or maybe `get_subnetwork_hex_identifier` would be better